### ISSUE TITLE
Question: Is there a cop that enforces the recommendation of 1 class per file?

### ISSUE THREAD

**Issue by 00dav00** [rubocop/rubocop#11519]
Checking the ruby style guide I see that there is a recommendation on having a single [class/module per file](https://rubystyle.guide/#one-class-per-file):

```
Aim to have just a single class/module per source file. Name the file name as the class/module, but replacing CapitalCase with snake_case.
```

I'm working on a project where we would like rubocop to help us enforce this rule but I can't find the right cop. Does anybody knows if there is currently a cop for this?

**Comment by fatkodima** [comment]
I was unable to find existing one, nor previous issues/PR related to it.

**Comment by bbatsov** [comment]
Yeah, seems we somehow missed this one, even if it's both useful and easy to implement.

**Comment by dvandersluis** [comment]
There are definitely going to be edge cases for this, for instance, how many classes are defined here?

```ruby
# lib/a/b/c.rb
class A
  class B
    class C
      def initialize
        # ...
      end
    end
  end
end
```

(regardless of the fact that classes within classes are not great 😆)

**Comment by bbatsov** [comment]
@dvandersluis We already have some code handling such cases for the `Style/Documentation` cop. Basically you should be able to tell apart "namespaces" from the actual implementation. I have some vague memory that there was also a cop checking whether a file name matches a class/module name. Or maybe this was just another idea as well.

**Comment by dduugg** [comment]
It'd be nice if this had some flexibility, perhaps a list of classes that can be subclassed without triggering violations (e.g. i think it's reasonable to subclass `StandardError` a few times in the same file, and presumably [rubocop](https://github.com/rubocop/rubocop/blob/master/lib/rubocop/server/errors.rb) does as well.)
